src/wagtailtrans/models.py

```diff
@@ -261,9 +261,7 @@ def force_parent_language(self, parent=None):
         if hasattr(parent, 'language'):
             if self.language != parent.language:
                 self.language = parent.language
-        elif get_wagtailtrans_setting('LANGUAGES_PER_SITE'):
-            site = parent.get_site()
-            self.language = site.sitelanguages.default_language
+
```

Comment:
This is only called when the parent has no language, i.e. for the root page in the tree. In that case the language for the page, which defaults to the global default language is overwritten with the sites default language.